File: RecylerViewMenu/app/src/main/java/com/example/recylerviewmenu/MainActivity.java
package com.example.recylerviewmenu;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    AppCompatButton btnList1, btnList2, btnGrid;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnGrid =findViewById(R.id.btngrid);
        btnList1 =findViewById(R.id.btnList1);
        btnList2 =findViewById(R.id.btnList2);
        recyclerView = findViewById(R.id.recylerView);

        ArrayList<Food> foods = new ArrayList<>();

        foods.add(new Food("Arabian Delight", R.drawable.arabian_delight));
        foods.add(new Food("Arabian Shawarma", R.drawable.arabian_shawarma));
        foods.add(new Food("Biriyani", R.drawable.biriyani));
        foods.add(new Food("Qorma", R.drawable.qorma));
        foods.add(new Food("Ravi Charga", R.drawable.ravi_charga));
        foods.add(new Food("KFC", R.drawable.kfc));
        foods.add(new Food("Saviour Foods", R.drawable.saviour_foods));

        FoodAdapter adapter = new FoodAdapter(this,foods);

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.HORIZONTAL);

        LinearLayoutManager linearLayoutManager1 = new LinearLayoutManager(this);

        GridLayoutManager gridLayoutManager = new GridLayoutManager(this, 2);

        btnList1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                recyclerView.setLayoutManager(linearLayoutManager);
            }
        });

        btnList2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                recyclerView.setLayoutManager(linearLayoutManager1);
            }
        });

        btnGrid.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                recyclerView.setLayoutManager(gridLayoutManager);
            }
        });
         recyclerView.setAdapter(adapter);
    }
}

File: RecylerViewkt/app/src/main/java/com/example/recylerviewkt/MainActivity.kt
package com.example.recylerviewkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val arrContact = ArrayList<ContactModel>()

        arrContact.add(ContactModel(R.drawable.images,"A", "0987654321"))
        arrContact.add(ContactModel(R.drawable.images,"B", "0887654321"))
        arrContact.add(ContactModel(R.drawable.images,"C", "0787654321"))
        arrContact.add(ContactModel(R.drawable.images,"D", "0687654321"))
        arrContact.add(ContactModel(R.drawable.images,"E", "0587654321"))
        arrContact.add(ContactModel(R.drawable.images,"F", "0487654321"))
        arrContact.add(ContactModel(R.drawable.images,"G", "0387654321"))
        arrContact.add(ContactModel(R.drawable.images,"H", "0287654321"))
        arrContact.add(ContactModel(R.drawable.images,"I", "0187654321"))
        arrContact.add(ContactModel(R.drawable.images,"J", "0087654321"))
        arrContact.add(ContactModel(R.drawable.images,"K", "0997654321"))
        arrContact.add(ContactModel(R.drawable.images,"L", "0887654321"))
        arrContact.add(ContactModel(R.drawable.images,"M", "0777654321"))
        arrContact.add(ContactModel(R.drawable.images,"N", "0667654321"))
        arrContact.add(ContactModel(R.drawable.images,"O", "0557654321"))
        arrContact.add(ContactModel(R.drawable.images,"P", "0447654321"))
        arrContact.add(ContactModel(R.drawable.images,"Q", "0337654321"))
        arrContact.add(ContactModel(R.drawable.images,"R", "0227654321"))
        arrContact.add(ContactModel(R.drawable.images,"S", "0117654321"))
        arrContact.add(ContactModel(R.drawable.images,"T", "0007654321"))
        arrContact.add(ContactModel(R.drawable.images,"U", "0999654321"))
        arrContact.add(ContactModel(R.drawable.images,"V", "0888654321"))
        arrContact.add(ContactModel(R.drawable.images,"W", "0777654321"))
        arrContact.add(ContactModel(R.drawable.images,"X", "0666654321"))
        arrContact.add(ContactModel(R.drawable.images,"Y", "0555654321"))
        arrContact.add(ContactModel(R.drawable.images,"Z", "0444654321"))

        recylerView.layoutManager= GridLayoutManager(this,3)
        val recylerAdapter = RecylerContactAdapter(this, arrContact)
        recylerView.adapter = recylerAdapter
    }

}